semgrep-core/src/parsing/pfff/java_to_generic.ml

```diff
@@ -421,6 +421,19 @@ and arguments v : G.argument list G.bracket = bracket (list argument) v
 
 and fix_op v = H.conv_incr v
 
+and resource (v : resource) : G.stmt =
+  match v with
+  | Left v ->
+      let ent, v = var_with_init v in
+      G.DefStmt (ent, G.VarDef v) |> G.s
+  | Right e -> G.ExprStmt (expr e, unsafe_fake "") |> G.s
+
+and resources (t1, v, t2) =
+  match v with
+  | [] -> G.Block (t1, [], t2) |> G.s
+  | [ v ] -> resource v
+  | _ :: _ -> G.Block (t1, list resource v, t2) |> G.s
```

Comment:
I think you could merge the cases `[]` and `_ :: _`. Do we need a special case for `[ v ]` ?

Reply:
I wasn't sure if it would be better to avoid the unnecessary Block, can take it out

Reply:
I would keep the translation trivial unless there is a good reason to save that `Block`. If you prefer to save the `Block` I would merge `[]` and `_ :: _` since they are the same. But this are just minor observations anyways. It's your call!